packages/b/bash/files/profile/50-prompt.sh

```diff
@@ -1,23 +1,53 @@
 # Begin /usr/share/defaults/etc/profile.d/50-prompt.sh
 
+# We cant use $SHELL since we need the *current* shell.
+# The user may switch to a different shell than their default
+# e.g. $SHELL is zsh but user switches to bash temporarily
+
+current_shell=$(readlink /proc/$$/exe)
+
 endchar="\$"
 if [ "$UID" = "0" ]; then
     endchar="#"
+elif [[ "$current_shell" =~ zsh ]]; then
+    endchar='%'
 fi
 
-FG="\[\033[38;5;081m\]"
-BG="\[\033[38;5;245m\]"
-AT="\[\033[38;5;245m\]"
-HCOLOR="\[\033[38;5;206m\]"
+# Prefer tput to ANSI codes since they work with bash and zsh
+# and are much easier to read
+grey=$(tput setaf 245)
+pink=$(tput setaf 206)
+cyan=$(tput setaf 81)
+reset_color=$(tput sgr0)
 
-PS1="${FG}\u${AT}@${HCOLOR}\H ${BG}\w ${FG}$endchar \[\e[0m\]"
+FG=$cyan
+DIR=$grey
+AT=$grey
+HCOLOR=$pink
+
+# bash and zsh handle prompt setup and character escaping differently
+
+if [[ "$current_shell" =~ bash ]]; then
+    PS1="\[${FG}\]\u\[${AT}\]@\[${HCOLOR}\]\H \[${DIR}\]\w \[${FG}\]$endchar \[${reset_color}\]"
+elif [[ "$current_shell" =~ zsh ]]; then
+    PS1="%{${FG}%}%n%{${AT}%}@%{${HCOLOR}%}%m %{${DIR}%}%1~ %{${FG}%}$endchar# ${reset_color}"
```

Comment:
Taken together with my suggestion above, the `#` after `$endchar` is no longer necessary.

Reply:
Fixed

Reply:
Reverted this. That # is necessary in zsh for the endchar to be written to the prompt. Without it, no % appears